### ISSUE TITLE
Duplicated tags in table

### ISSUE THREAD

**Issue by mbkv** [mautic/mautic#6057]
**Please DO NOT report security vulnerabilities here. Send them to security@mautic.com instead.**

What type of report is this:

| Q  | A
| ---| ---
| Bug report? | y
| Feature request? | n
| Enhancement? | n

## Description:
When tags gets created with a special character (IE: `"` or `'`), tags become duplicated

I figured out the issue already, but I'm not sure if my fix would break anything elsewhere
The issue is located here
https://github.com/mautic/mautic/blob/2.13.1/app/bundles/LeadBundle/Model/LeadModel.php#L1887

The tag gets searched here, and then on line 1903, the tag class gets created. Inside the constructor, the tag gets validated once more
https://github.com/mautic/mautic/blob/2.13.1/app/bundles/LeadBundle/Entity/Tag.php#L37

Since the value it searches for and the value that gets created aren't the same, every single time you put tag with a special character, it'll get created again

It looks like the tag was supposed to be validated beforehand with the array_walk on 1884, but I don't think this actually works because it's not being taken by reference inside the `InputHelper::clean` function

## If a bug:

| Q   | A
| --- | ---
| Mautic version | 2.13.1
| PHP version | 7.0

### Steps to reproduce:
1. Create a tag with the tracking script with a special character `"`
2. Use Tor or some proxy service to go to a new IP and do it again with the same tag
3. Two rows gets created in the database, both with the same tag
 
### Log errors: 

Since there's only debug logs near the issue, this doesn't have any logs

_Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable._

**Comment by mbitokhov** [comment]
Just for some evidence, this is the output of
```sql
SELECT
  id,
  tag
FROM `lead_tags`
WHERE tag like '%refrig%'
```
![2018-05-07 12_38_55-sql command - db - adminer](https://user-images.githubusercontent.com/9276556/39713142-ac8d13c4-51f3-11e8-844f-201a0098e39b.png)

**Comment by npracht** [comment]
Label: Bug

**Comment by npracht** [comment]
@mbkv can you please test #7741 ?

**Comment by npracht** [comment]
closed by #7741
